server/channels/store/sqlstore/group_store.go

```diff
@@ -1284,34 +1284,42 @@ func (s *SqlGroupStore) ChannelMembersToRemove(channelID *string) ([]*model.Chan
 }
 
 func (s *SqlGroupStore) groupsBySyncableBaseQuery(st model.GroupSyncableType, t selectType, syncableID string, opts model.GroupSearchOpts) sq.SelectBuilder {
-	selectStrs := map[selectType]string{
-		selectGroups:      "UserGroups.*, gs.SchemeAdmin AS SyncableSchemeAdmin",
-		selectCountGroups: "COUNT(*)",
+	var query sq.SelectBuilder
+	switch t {
+	case selectGroups:
+		query = s.userGroupsSelectQuery.
+			Column("gs.SchemeAdmin AS SyncableSchemeAdmin")
+	case selectCountGroups:
+		query = s.getQueryBuilder().
+			Select("COUNT(*)").
+			From("UserGroups")
 	}
 
-	var table string
-	var idCol string
 	if st == model.GroupSyncableTypeTeam {
-		table = "GroupTeams"
-		idCol = "TeamId"
+		query = query.
+			Join("GroupTeams gs ON gs.GroupId = UserGroups.Id").
```

Comment:
As above, we now invert the join, being explicit about the table instead of using variables. Note we use the same alias `gs` regardless of whether `GroupTeams` or `GroupChannels` is the chosen table.